src/pe/authenticode.rs

```diff
@@ -0,0 +1,117 @@
+// Reference:
+//   https://learn.microsoft.com/en-us/windows-hardware/drivers/install/authenticode
+//   https://download.microsoft.com/download/9/c/5/9c5b2167-8017-4bae-9fde-d599bac8184a/Authenticode_PE.docx
+
+// Authenticode works by omiting sections of the PE binary from the digest
+// those sections are:
+//   - checksum
+//   - data directory entry for certtable
+//   - certtable
+
+use alloc::{boxed::Box, vec::Vec};
+use core::ops::Range;
+
+use super::PE;
+
+impl PE<'_> {
+    /// [`authenticode_ranges`] returns the various ranges of the binary that are relevant for
+    /// signature.
+    pub fn authenticode_ranges(&self) -> ExcludedAuthenticodeSectionsIter<'_> {
+        ExcludedAuthenticodeSectionsIter {
+            pe: self,
+            state: IterState::default(),
+        }
+    }
+
+    /// [`authenticode_slice`] is intended for convenience when signing a binary with a PKCS#11
+    /// interface (HSM interface).
+    /// Some algorithms (RSA-PKCS at least) for signature require the non-prehashed slice to be provided.
+    pub fn authenticode_slice(&self) -> Box<[u8]> {
```

Comment:
is this really useful to have here? I hesitate to add `pub` functions since they become maintenance burden/have to be watched for breaking changes, etc.